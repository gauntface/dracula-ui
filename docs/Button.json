{
  "description": "The Button component triggers actions, behaviors, or events based\non user input.",
  "displayName": "Button",
  "methods": [],
  "props": {
    "themeColor": {
      "defaultValue": null,
      "description": "A Dracula UI theme color for the Button.",
      "name": "themeColor",
      "parent": {
        "fileName": "src/components/Button/Button.tsx",
        "name": "ButtonProps"
      },
      "required": false,
      "type": {
        "name": "\"black\" | \"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
      }
    },
    "variant": {
      "defaultValue": null,
      "description": "Controls the Button style:\n`normal` -> Default solid Button with background color.\n`outline` -> A subtle variation of the Button component with a softer background color that highlights the action text.\n`ghost` -> A less prominent variation of the Button component that highlights hover interactions.",
      "name": "variant",
      "parent": {
        "fileName": "src/components/Button/Button.tsx",
        "name": "ButtonProps"
      },
      "required": false,
      "type": {
        "name": "\"normal\" | \"outline\" | \"ghost\" | undefined"
      }
    },
    "disabled": {
      "defaultValue": null,
      "description": "Controls the Button state. Mirrors the HTMLButtonElement `disabled` property.",
      "name": "disabled",
      "parent": {
        "fileName": "src/components/Button/Button.tsx",
        "name": "ButtonProps"
      },
      "required": false,
      "type": {
        "name": "boolean | undefined"
      }
    }
  }
}
