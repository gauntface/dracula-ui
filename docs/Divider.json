{
  "description": "Dividers are horizontal lines used to separate semantic blocks of\ncontent or UI patterns.",
  "displayName": "Divider",
  "methods": [],
  "props": {
    "color": {
      "defaultValue": null,
      "description": "The Dracula UI color for the Divider.",
      "name": "color",
      "parent": {
        "fileName": "src/components/Divider/Divider.tsx",
        "name": "DividerProps"
      },
      "required": true,
      "type": {
        "name": "\"black\" | \"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\""
      }
    }
  }
}
