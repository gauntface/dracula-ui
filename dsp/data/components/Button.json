{
  "dsp_spec_version": "0.0.1",
  "last_updated_by": "System",
  "last_updated": "2020-10-04T01:34:14.362Z",
  "entities": [
    {
      "class": "component",
      "type": "page",
      "id": "Button",
      "name": "Button",
      "last_updated": "2020-10-04T01:34:14.362Z",
      "last_updated_by": "System",
      "description": "The Button component triggers actions, behaviors, or events based\non user input.",
      "related_entity_ids": [],
      "tags": ["component"],
      "snippets": {
        "trigger": "Button",
        "languages": {
          "javascript": "<div>\n  <Box style={{...}}>\n    <Button color=\"purple\">\n      normal\n    </Button>\n  </Box>\n</div>",
          "html": "<div style=\"padding: 3rem\" class=\"drac-box\">\n  <button class=\"drac-btn drac-bg-purple drac-btn\" color=\"purple\">\n    <span class=\"drac-text drac-text drac-text drac-text-black\">normal</span>\n  </button>\n</div>\n"
        }
      }
    }
  ]
}
