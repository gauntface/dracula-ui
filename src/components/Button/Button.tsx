import React, { HTMLAttributes } from 'react'
import { colors as backgroundColors } from '@/base/colors'
import cx from 'classnames'
import { Text, TextColors } from '@/components/Typography/Text'

const variants = {
  normal: 'drac-btn',
  outline: 'drac-btn-outline',
  ghost: 'drac-btn-ghost'
}

/** Button Props */
export interface ButtonProps extends HTMLAttributes<HTMLButtonElement> {
  /** A Dracula UI theme color for the Button. */
  color?: keyof typeof backgroundColors

  /**
   * Controls the Button style:
   * `normal` -> Default solid Button with background color.
   * `outline` -> A subtle variation of the Button component with a softer background color that highlights the action text.
   * `ghost` -> A less prominent variation of the Button component that highlights hover interactions.
   */
  variant?: keyof typeof variants

  /**
   * Controls the Button state. Mirrors the HTMLButtonElement `disabled` property.
   */
  disabled?: boolean
}

/**
 * The Button component triggers actions, behaviors, or events based
 * on user input.
 */
export const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
  const isOutline = props.variant === 'outline'
  const isGhost = props.variant === 'ghost'
  const overrideTextColor = isOutline || isGhost

  const textColorClass = overrideTextColor
    ? TextColors[props.color ?? 'green']
    : undefined

  let backgroundClass = backgroundColors[props.color ?? 'green']
  if (isGhost) {
    backgroundClass = `${backgroundClass}-transparent`
  }

  const classes = cx(
    'drac-btn',
    backgroundClass,
    variants[props.variant ?? 'normal'],
    textColorClass
  )

  return (
    <button className={classes} {...props}>
      <Text color={overrideTextColor ? props.color : undefined}>
        {props.children}
      </Text>
    </button>
  )
}

Button.displayName = 'Button'
