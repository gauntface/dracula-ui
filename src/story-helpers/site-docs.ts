import {
  renderSnapshot,
  SnapshotBuilder
} from '@/story-helpers/render-component'
import React from 'react'

// import * as serializer from 'jest-snapshot-serializer-raw'

interface Documentation {
  basicUsage: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder>
}

export function siteDocs<T>(
  componentClass: React.FunctionComponent<T>,
  documentation: Documentation
) {
  describe(`Site: ${componentClass.displayName ?? componentClass.name}`, () => {
    test('Basic Usage', () => {
      const variation = documentation.basicUsage()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp(),
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()
    })

    const variations = documentation.variations()

    variations.forEach(variation => {
      test(variation.title, () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp(),
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()
      })
    })
  })
}
