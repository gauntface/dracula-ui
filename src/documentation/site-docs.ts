import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from '../story-helpers/render-component'
import React from 'react'
import fs from 'fs-extra'
import { pretty } from '../story-helpers/pretty'
import { getDocGen } from '../documentation/docgen/doc-generator'
import { toDSP } from '../documentation/dsp/component-generator'

interface Documentation {
  basicUsage: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder>
}

export function siteDocs<T>(
  componentClass: React.FunctionComponent<T>,
  documentation: Documentation
) {
  const name = componentClass.displayName ?? componentClass.name

  describe(`Site: ${name}`, () => {
    let examples: Record<string, ComponentExample> = {}

    afterAll(async () => {
      const docGen = getDocGen(name)
      const dsp = toDSP(name, examples, docGen)
      const dspPath = `${process.cwd()}/dsp/data/components/${name}.json`
      await fs.writeFile(dspPath, pretty(JSON.stringify(dsp), 'json'))
    })

    test('Basic Usage', () => {
      const variation = documentation.basicUsage()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()
      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations = documentation.variations()

    variations.forEach((variation) => {
      test(variation.title, () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
