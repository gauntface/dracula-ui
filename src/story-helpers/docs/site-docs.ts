import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from '@/story-helpers/render-component'
import React from 'react'
import fs from 'fs'
import { pretty } from '@/story-helpers/pretty'

interface Documentation {
  basicUsage: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder>
}

export function siteDocs<T>(
  componentClass: React.FunctionComponent<T>,
  documentation: Documentation
) {
  const name = componentClass.displayName ?? componentClass.name

  describe(`Site: ${name}`, () => {
    let examples: Record<string, ComponentExample> = {}

    afterAll(() => {
      const path = `${process.cwd()}/examples/${name}.json`
      fs.writeFileSync(path, pretty(JSON.stringify(examples), 'json'))
    })

    test('Basic Usage', () => {
      const variation = documentation.basicUsage()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()
      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations = documentation.variations()

    variations.forEach((variation) => {
      test(variation.title, () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
