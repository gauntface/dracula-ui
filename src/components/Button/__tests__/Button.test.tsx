import { Button } from '@/components/Button/Button'
import {
  buildSnapshot,
  SnapshotBuilder
} from '@/story-helpers/render-component'
import { siteDocs } from '@/story-helpers/docs/site-docs'
import {
  ButtonNormal,
  ButtonVariants,
  ButtonColors
} from '@/components/Button/__stories__/Button.stories'

siteDocs(Button, {
  basicUsage() {
    return buildSnapshot('Usage', ButtonNormal)
  },
  variations() {
    return [
      buildSnapshot('Variants', ButtonVariants),
      buildSnapshot('Colors', ButtonColors)
    ] as SnapshotBuilder[]
  }
})
