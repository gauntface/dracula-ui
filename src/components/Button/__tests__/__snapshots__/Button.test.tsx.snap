// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: Button Basic Usage 1`] = `
Object {
  "docs": "",
  "html": "<button class=\\"drac-btn drac-bg-purple\\" color=\\"purple\\">Dracula</button>
",
  "react": "<Button color=\\"purple\\">Dracula</Button>;
",
  "title": "Usage",
}
`;

exports[`Site: Button Colors 1`] = `
Object {
  "docs": "",
  "html": "<div class=\\"drac-box\\">
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-white\\" color=\\"white\\" style=\\"margin: 1rem\\">
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-white drac-btn-outline drac-text-white\\"
      color=\\"white\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-white-transparent drac-btn-ghost drac-text-white\\"
      color=\\"white\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-black\\" color=\\"black\\" style=\\"margin: 1rem\\">
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-black drac-btn-outline drac-text-black\\"
      color=\\"black\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-black-transparent drac-btn-ghost drac-text-black\\"
      color=\\"black\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-black-secondary\\"
      color=\\"blackSecondary\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-black-secondary drac-btn-outline drac-text-black-secondary\\"
      color=\\"blackSecondary\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-black-secondary-transparent drac-btn-ghost drac-text-black-secondary\\"
      color=\\"blackSecondary\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-black-light\\"
      color=\\"blackLight\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-black-light drac-btn-outline drac-text-black-light\\"
      color=\\"blackLight\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-black-light-transparent drac-btn-ghost drac-text-black-light\\"
      color=\\"blackLight\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-cyan\\" color=\\"cyan\\" style=\\"margin: 1rem\\">
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-cyan drac-btn-outline drac-text-cyan\\"
      color=\\"cyan\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-cyan-transparent drac-btn-ghost drac-text-cyan\\"
      color=\\"cyan\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-green\\" color=\\"green\\" style=\\"margin: 1rem\\">
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-green drac-btn-outline drac-text-green\\"
      color=\\"green\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-green-transparent drac-btn-ghost drac-text-green\\"
      color=\\"green\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-orange\\" color=\\"orange\\" style=\\"margin: 1rem\\">
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-orange drac-btn-outline drac-text-orange\\"
      color=\\"orange\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-orange-transparent drac-btn-ghost drac-text-orange\\"
      color=\\"orange\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-pink\\" color=\\"pink\\" style=\\"margin: 1rem\\">
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-pink drac-btn-outline drac-text-pink\\"
      color=\\"pink\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-pink-transparent drac-btn-ghost drac-text-pink\\"
      color=\\"pink\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-purple\\" color=\\"purple\\" style=\\"margin: 1rem\\">
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-purple drac-btn-outline drac-text-purple\\"
      color=\\"purple\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-purple-transparent drac-btn-ghost drac-text-purple\\"
      color=\\"purple\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-red\\" color=\\"red\\" style=\\"margin: 1rem\\">
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-red drac-btn-outline drac-text-red\\"
      color=\\"red\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-red-transparent drac-btn-ghost drac-text-red\\"
      color=\\"red\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-yellow\\" color=\\"yellow\\" style=\\"margin: 1rem\\">
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-yellow drac-btn-outline drac-text-yellow\\"
      color=\\"yellow\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-yellow-transparent drac-btn-ghost drac-text-yellow\\"
      color=\\"yellow\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-purple-cyan\\"
      color=\\"purpleCyan\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-purple-cyan drac-btn-outline drac-text-purple-cyan\\"
      color=\\"purpleCyan\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-purple-cyan-transparent drac-btn-ghost drac-text-purple-cyan\\"
      color=\\"purpleCyan\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-yellow-pink\\"
      color=\\"yellowPink\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-yellow-pink drac-btn-outline drac-text-yellow-pink\\"
      color=\\"yellowPink\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-yellow-pink-transparent drac-btn-ghost drac-text-yellow-pink\\"
      color=\\"yellowPink\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-cyan-green\\"
      color=\\"cyanGreen\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-cyan-green drac-btn-outline drac-text-cyan-green\\"
      color=\\"cyanGreen\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-cyan-green-transparent drac-btn-ghost drac-text-cyan-green\\"
      color=\\"cyanGreen\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-pink-purple\\"
      color=\\"pinkPurple\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-pink-purple drac-btn-outline drac-text-pink-purple\\"
      color=\\"pinkPurple\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-pink-purple-transparent drac-btn-ghost drac-text-pink-purple\\"
      color=\\"pinkPurple\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
  <div style=\\"display: flex; justify-content: space-around\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-animated\\"
      color=\\"animated\\"
      style=\\"margin: 1rem\\"
    >
      Normal</button
    ><button
      class=\\"drac-btn drac-bg-animated drac-btn-outline drac-text-animated\\"
      color=\\"animated\\"
      style=\\"margin: 1rem\\"
      variant=\\"outline\\"
    >
      Outline</button
    ><button
      class=\\"drac-btn drac-bg-animated-transparent drac-btn-ghost drac-text-animated\\"
      color=\\"animated\\"
      style=\\"margin: 1rem\\"
      variant=\\"ghost\\"
    >
      Ghost
    </button>
  </div>
</div>
",
  "react": "<Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"white\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"white\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"white\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"black\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"black\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"black\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"blackSecondary\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"blackSecondary\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"blackSecondary\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"blackLight\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"blackLight\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"blackLight\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"cyan\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"cyan\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"cyan\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"green\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"green\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"green\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"orange\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"orange\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"orange\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"pink\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"pink\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"pink\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"purple\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"purple\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"purple\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"red\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"red\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"red\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"yellow\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"yellow\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"yellow\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"purpleCyan\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"purpleCyan\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"purpleCyan\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"yellowPink\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"yellowPink\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"yellowPink\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"cyanGreen\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"cyanGreen\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"cyanGreen\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"pinkPurple\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"pinkPurple\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"pinkPurple\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
  <Box style={{ display: \\"flex\\", justifyContent: \\"space-around\\" }}>
    <Button color=\\"animated\\" style={{ margin: \\"1rem\\" }}>
      Normal
    </Button>
    <Button color=\\"animated\\" style={{ margin: \\"1rem\\" }} variant=\\"outline\\">
      Outline
    </Button>
    <Button color=\\"animated\\" style={{ margin: \\"1rem\\" }} variant=\\"ghost\\">
      Ghost
    </Button>
  </Box>
</Box>;
",
  "title": "Colors",
}
`;

exports[`Site: Button Sizes 1`] = `
Object {
  "docs": "",
  "html": "<div class=\\"drac-box\\">
  <div style=\\"margin-bottom: 10px\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-purple drac-btn-lg\\" color=\\"purple\\">
      large
    </button>
  </div>
  <div style=\\"margin-bottom: 10px\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-purple\\" color=\\"purple\\">medium</button>
  </div>
  <div style=\\"margin-bottom: 10px\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-purple drac-btn-sm\\" color=\\"purple\\">
      small
    </button>
  </div>
  <div style=\\"margin-bottom: 10px\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-purple drac-btn-xs\\" color=\\"purple\\">
      xsmall
    </button>
  </div>
</div>
",
  "react": "<Box>
  <Box style={{ marginBottom: 10 }}>
    <Button color=\\"purple\\" size=\\"large\\">
      large
    </Button>
  </Box>
  <Box style={{ marginBottom: 10 }}>
    <Button color=\\"purple\\" size=\\"medium\\">
      medium
    </Button>
  </Box>
  <Box style={{ marginBottom: 10 }}>
    <Button color=\\"purple\\" size=\\"small\\">
      small
    </Button>
  </Box>
  <Box style={{ marginBottom: 10 }}>
    <Button color=\\"purple\\" size=\\"xsmall\\">
      xsmall
    </Button>
  </Box>
</Box>;
",
  "title": "Sizes",
}
`;

exports[`Site: Button Variants 1`] = `
Object {
  "docs": "",
  "html": "<div class=\\"drac-box\\">
  <div style=\\"margin-bottom: 10px\\" class=\\"drac-box\\">
    <button class=\\"drac-btn drac-bg-purple\\" variant=\\"normal\\" color=\\"purple\\">
      normal
    </button>
  </div>
  <div style=\\"margin-bottom: 10px\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-purple drac-btn-outline drac-text-purple\\"
      variant=\\"outline\\"
      color=\\"purple\\"
    >
      outline
    </button>
  </div>
  <div style=\\"margin-bottom: 10px\\" class=\\"drac-box\\">
    <button
      class=\\"drac-btn drac-bg-purple-transparent drac-btn-ghost drac-text-purple\\"
      variant=\\"ghost\\"
      color=\\"purple\\"
    >
      ghost
    </button>
  </div>
</div>
",
  "react": "<Box>
  <Box style={{ marginBottom: 10 }}>
    <Button variant=\\"normal\\" color=\\"purple\\">
      normal
    </Button>
  </Box>
  <Box style={{ marginBottom: 10 }}>
    <Button variant=\\"outline\\" color=\\"purple\\">
      outline
    </Button>
  </Box>
  <Box style={{ marginBottom: 10 }}>
    <Button variant=\\"ghost\\" color=\\"purple\\">
      ghost
    </Button>
  </Box>
</Box>;
",
  "title": "Variants",
}
`;
